lib/components/AutoSuggest/AutoSuggest.tsx

```diff
@@ -328,6 +335,10 @@ const AutoSuggestInput = <PayloadType extends unknown>({
 								event.preventDefault();
 								if (typeof onChange === 'function')
 									onChange(suggestions[state.highlightIndex]);
+								if (typeof onSelection === 'function')
```

Comment:
Could these conditionals be done as a switch with multiple options (onChange, onSelection and so forth)?